internal/notifier/client.go

```diff
@@ -32,29 +32,90 @@ import (
 	"github.com/hashicorp/go-retryablehttp"
 )
 
-type requestOptFunc func(*retryablehttp.Request)
+type postOption struct {
+	proxy             string
+	certPool          *x509.CertPool
+	requestModifiers  []requestModifier
+	responseValidator func(*http.Response) bool
+}
+type requestModifier func(*retryablehttp.Request)
+
+func postMessage(ctx context.Context, address string, payload interface{}, opt *postOption) error {
```

Comment:
I still prefer receiving the options as a variadic slice of functional options:

```suggestion
func postMessage(ctx context.Context, address string, payload interface{}, opts ...postOptions) error {
```

Where:

```go
type postOption func(*postOptions)
```

Let's define also `with<Something>()` helper functions for passing the options inside the providers. We recently refactor the factory constructor to adhere to this style, let's follow it everywhere :pray: